src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -1121,10 +1095,7 @@ private BoundExpression MakeLiftedUserDefinedConversionConsequence(BoundCall cal
                 return BoundConversion.Synthesized(syntax, rewrittenOperand, new Conversion(conversionKind, method, false), @checked, explicitCastInCode, constantValueOpt, rewrittenType);
             }
 
-            var rewrittenCall =
-                _inExpressionLambda && oldNode != null
```

Comment:
This condition was always false in the compiler because when `_inExpressionLambda` is `true` the code above unconditionally returns.